### PR TITLE
Enhancements

### PR DESCRIPTION
**Pull request by romkatsu**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | Enhancements

### CODE REVIEW

**Review by rustamwin — APPROVED** [review]

File: src/MatchingResult.php [hunk 1]
```diff
@@ -16,10 +18,6 @@ final class MatchingResult implements MiddlewareInterface
     private array $methods = [];
     private ?DispatcherInterface $dispatcher = null;
 
-    private function __construct()
```
Comment (yiiliveext) [inline src/MatchingResult.php:19]: Why did you remove the private constructor? This class should only be instantiated by factory methods.

Comment (romkatsu) [reply src/MatchingResult.php]: Thanks. Fixed.

File: src/RouteCollection.php [hunk 2]
```diff
@@ -22,8 +22,6 @@ final class RouteCollection implements RouteCollectionInterface
     /**
      * RouteCollection constructor.
      * @param RouteCollectorInterface $collector
-     * @param string $prefix
-     * @param bool $buildRoutes
```
Comment (roxblnfk) [inline src/RouteCollection.php:26]: You can delete this block entirely

Comment (roxblnfk) [inline src/RouteCollection.php:26]: You can delete this block entirely

**Review by xepozz — APPROVED** [review]

**Review by yiiliveext — APPROVED** [review]

**Review by samdark — APPROVED** [review]
